### PR TITLE
1616: Improve error message when an author isn't found in GitLab

### PR DESCRIPTION
**Pull request by zhaosongzs**
Improved the error message in GitLabMergeRequest:author

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-1616](https://bugs.openjdk.org/browse/SKARA-1616): Improve error message when an author isn't found in GitLab

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara pull/1390/head:pull/1390` \
`$ git checkout pull/1390`

Update a local copy of the PR: \
`$ git checkout pull/1390` \
`$ git pull https://git.openjdk.org/skara pull/1390/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1390`

View PR using the GUI difftool: \
`$ git pr show -t 1390`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1390.diff">https://git.openjdk.org/skara/pull/1390.diff</a>

</details>

### CODE REVIEW

File: forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabMergeRequest.java [hunk 1]
```diff
@@ -77,7 +77,13 @@ public String id() {
 
     @Override
     public HostUser author() {
-        return repository.forge().user(json.get("author").get("username").asString()).get();
+        var username = json.get("author").get("username").asString();
+        var author = repository.forge().user(username);
+        if (author.isPresent()) {
+            return author.get();
+        } else {
+            throw new IllegalArgumentException("Unknown username: " + username);
```
Comment (erikj79) [inline forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabMergeRequest.java:85]: I don't think IllegalArgumentException is correct here. It's meant to be thrown when a method receives a bad argument, but this method doesn't take arguments. We don't really have a reason to define a new exception type either, so a plain `RuntimeException` is probably better, at least for now.

I'm usually for short and concise messages, but in this case I think we can add some more details. Something like `"Author of merge request unknown: " + username`

Comment (zhaosongzs) [reply forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabMergeRequest.java:85]: Ok, I also thought about the type of exception yesterday. I was thinking that the username is an argument of `user(username)`, but throw exception in method `user` is a little complex. 

I will change it to RuntimeException and make the error message have more details.

**Review by erikj79 — APPROVED** [review]

**Review by lgxbslgx — APPROVED** [review]

**Pull request closed without merging**